presto-docs/src/main/sphinx/kerberos/hive.rst

```diff
@@ -0,0 +1,162 @@
+=====================
+Hive Kerberos Support
+=====================
+
+The :doc:`/connector/hive` supports connecting to an instance of Hive running on
+a Hadoop cluster that has Kerberos authentication enabled. There are two
+services on the Hadoop cluster that can be configured to require Kerberos
+authentication:
+
+* The Hive metastore Thrift service
+* The Hadoop Distributed File System (HDFS)
+
+Access to these services by the Hive connector is configured in the properties
+file that contains the general Hive connector configuration.
+
+Hive Metastore Thrift Service
+-----------------------------
+
+In a Kerberized Hadoop cluster, access to the Hive metastore Thrift service is
+done using the Simple Authentication and Security Layer (SASL). Kerberos
+authentication for the metastore is configured in the connector's properties
+file using the following properties:
+
+================================================== ============================================================ ==========
+Property Name                                      Description                                                  Default
+================================================== ============================================================ ==========
+``hive.metastore.sasl.enabled``                    When set to ``true`` the Hive connector will connect to the  ``false``
+                                                   Hive metastore Thrift service using SASL and authenticate
+                                                   with Kerberos. When using the default value of ``false``,
+                                                   no Kerberos authentication is disabled and no other
+                                                   properties need to be configured.
+
+``hive.metastore.principal``                       The Kerberos principal of the Hive metastore. The Presto     
+                                                   coordinator will use this to authenticate the Hive
+                                                   metastore.
+
+``hive.metastore.presto.principal``                The Kerberos principal that Presto will use when connecting
```

Comment:
Lukasz said we aren't going to have impersonation right now, but we need to clearly document this because it, in a way, subverts some possible Hadoop security.